Source/Models/APIClient.swift

```diff
@@ -4,46 +4,35 @@ import Result
 
 public struct APIClient {
   private let requestPerformer: RequestPerformer
+  public let deserializer: Deserializer
```

Comment:
As in, I'd prefer we _remove_ `deserializer` from the `Client` protocol and make this `private`. If there is a need to make these `public` down the line, that will be super easy, where as marking these as `private` later will be more difficult.